tools/build_executable/build_executable.py

```diff
@@ -0,0 +1,168 @@
+        else:
+            # see https://docs.anaconda.com/anaconda/user-guide/faq
+            conda_script = os.path.expanduser(os.path.join("~", "anaconda3", "etc", "profile.d", "conda.sh"))
+
+        cmd_prefix = f". {conda_script}; conda activate {args.conda_env}; "
+
+    else:
+        assert False
+
+    # once we know target_platform, set our default pak file src and dest
+    pak_file_src  = os.path.realpath(os.path.join(args.paks_dir, "kujiale_0000_" + target_platform + ".pak"))
+    pak_file_dest = os.path.realpath(os.path.join(paks_dir_dest, "kujiale_0000_" + target_platform + ".pak"))
+
+    # We need to remove this temp dir (created by the Unreal build process) because it contains paths from previous builds.
+    # If we don't do this step, we will get many warnings during this build:
+    #     Warning: Unable to generate long package name for path/to/previous/build/Some.uasset because FilenameToLongPackageName failed to convert
+    #     'path/to/previous/build/Some.uasset'. Attempt result was '../../../../../../path/to/previous/build/path/to/previous/build/Some', but the
+    #     path contains illegal characters '.'
+    if os.path.exists(unreal_tmp_dir):
+        print(f"[SPEAR | build_executable.py] Unreal Engine cache directory exists, removing: {unreal_tmp_dir}")
+        shutil.rmtree(unreal_tmp_dir)
+
+    if not args.skip_clone_github_repo:
+        
+        # remove our temporary repository directory to ensure a clean build
+        if os.path.exists(repo_dir):
+            print(f"[SPEAR | build_executable.py] Repository exists, removing: {repo_dir}")
+            shutil.rmtree(repo_dir)
+            os.makedirs(repo_dir)
+
+        # clone repo with submodules
+        cmd = ["git", "clone", "--recurse-submodules", "https://github.com/isl-org/spear", repo_dir]
+        print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+        subprocess.run(cmd, check=True)
+
+        # reset to a specific commit ID
+        if args.commit_id:
+            cmd = ["git", "reset", "--hard", args.commit_id]
+            print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+            subprocess.run(cmd, check=True)
+
+    if not args.skip_build_third_party_libs:
+
+        # build third-party libs
+        cmd = [
+            "python",
+            os.path.join("..", "build_third_party_libs.py"),
+            "--third_party_dir", third_party_dir,
+            "--num_parallel_jobs", f"{args.num_parallel_jobs}"
+        ]
+        print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+        subprocess.run(cmd, check=True)
+
+    # create symbolic links (need shell=True because we want to run in a specific anaconda env)
+    cmd = cmd_prefix + f"python {os.path.join('..', 'create_symbolic_links.py')} --unreal_engine_dir {args.unreal_engine_dir} --unreal_project_dir {unreal_project_dir} --unreal_plugins_dir {unreal_plugins_dir} --third_party_dir {third_party_dir}"
```

Comment:
On Windows, if your `args.unreal_engine_dir` contains spaces in the path (usually does in default installations), something like `C:\Epic Games\Program Files\`, this throws an error.

We can instead write it as
```python
cmd = cmd_prefix + f'python {os.path.join("..", "create_symbolic_links.py")} --unreal_engine_dir "{args.unreal_engine_dir}" --unreal_project_dir {unreal_project_dir} --unreal_plugins_dir {unreal_plugins_dir} --third_party_dir {third_party_dir}'
```

Reply:
Good catch. I just fixed this. We use `args.unreal_engine_dir` in several of our `tools` now, but I think this is the only place where this issue will occur.